src/object/cli_obj.c

```diff
@@ -1012,13 +1012,17 @@ obj_shard_tgts_query(struct dc_object *obj, uint32_t map_ver, uint32_t shard,
 	shard_tgt->st_shard	= shard;
 	shard_tgt->st_shard_id	= obj_shard->do_id.id_shard;
 	shard_tgt->st_tgt_idx	= obj_shard->do_target_idx;
-	if (obj_auxi->cond_modify && (obj_shard->do_rebuilding || obj_shard->do_reintegrating))
+	if ((obj_auxi->cond_modify || obj_is_modification_opc(obj_auxi->opc)) &&
```

Comment:
Say one object S1 is created and written some records,  then adds new ranks/targets, then do punch during extending, which then will punch on both "new" and "old" shards, but the shards only exist on the "old" shard, so the punch will fail with "-ENOEXIST" on the "new" shard.

So the solution here is that we set the DELAY flags on the reintegrating(new) shard, so during server forward, it will send "punch" to old/healthy shard first, then it will do create/punch by special flag on the reintegrating(DELAY) shard, instead of only punch.

Reply:
OK, as discussed by skype, here should be unnecessary for normal update.